sql/logictest: TestLogic failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/445ec411db8cde8cc090c7ffcd7ac70be9b21954

Parameters:
```
TAGS=
GOFLAGS=-parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLogic PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1521061&tab=buildLog

```
=== RUN   TestLogic
--- FAIL: TestLogic (585.93s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestLogic393066543
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestLogic/fakedist-disk/upsert
=== PAUSE TestLogic/fakedist-disk/upsert
=== CONT  TestLogic/fakedist-disk/upsert
        --- FAIL: TestLogic/fakedist-disk/upsert (0.81s)
=== RUN   TestLogic/fakedist-disk
    --- FAIL: TestLogic/fakedist-disk (0.46s)
=== RUN   TestLogic/fakedist-disk/upsert/notstrict
rewrote: CREATE TABLE abc (a INT8, b INT8, c INT8 DEFAULT 7, PRIMARY KEY (a, b), INDEX y (b), CONSTRAINT z UNIQUE (c), FAMILY (c));
rewrote: CREATE TABLE upsert_returning (a INT8 PRIMARY KEY, b INT8, c INT8, d INT8 DEFAULT -1, FAMILY (c), FAMILY (d), FAMILY (b));
--- done: testdata/logic_test/upsert with config fakedist-disk: 75 tests, 2 failures
            --- FAIL: TestLogic/fakedist-disk/upsert/notstrict (0.16s)
                logic.go:1655: 
                     
                    testdata/logic_test/upsert:339: SELECT b FROM upsert_returning WHERE a = 1
                    expected:
                        1
                        
                    but found (query options: "") :
                        1
                        NULL
                        
                    
                logic.go:1307: 
                     testdata/logic_test/upsert:344: too many errors encountered, skipping the rest of the input

```

SHA: https://github.com/cockroachdb/cockroach/commits/1179c76e34d057259854d88c7948830d2b4b05cd

Parameters:
```
TAGS=
GOFLAGS=-parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLogic PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1607268&tab=buildLog

```

panic: flow: unexpected 71680 leftover bytes

goroutine 1969872 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.ReportOrPanic(0x41e3140, 0xc0162192c0, 0xc00002ea80, 0xc00d02e240, 0x25, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:554 +0x227
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc0032f55e0, 0x41e3140, 0xc0162192c0, 0xc00e530901)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:397 +0x29e
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:384
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*EvalContext).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/eval.go:2795
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cleanup(0xc0127b5a20, 0x41e3140, 0xc0162192c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:437 +0x163
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowScheduler).runFlowNow.func2(0x42449a0, 0xc0114c0a70, 0x41e3140, 0xc0162192c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow_scheduler.go:104 +0x59
created by github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowScheduler).runFlowNow
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow_scheduler.go:102 +0x280

goroutine 1 [chan receive, 3 minutes]:
testing.(*T).Run(0xc000892200, 0x391f4b6, 0x9, 0x3a79f68, 0x712016)
	/usr/local/go/src/testing/testing.go:917 +0x381
testing.runTests.func1(0xc000892100)
	/usr/local/go/src/testing/testing.go:1157 +0x78
testing.tRunner(0xc000892100, 0xc00071be10)
	/usr/local/go/src/testing/testing.go:865 +0xc0
testing.runTests(0xc00027a4c0, 0x6473a80, 0x4, 0x4, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x2a9
testing.(*M).Run(0xc0001ac480, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x162
github.com/cockroachdb/cockroach/pkg/sql/logictest.TestMain(0xc0001ac480)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/main_test.go:33 +0xb8
main.main()
	_testmain.go:46 +0x13e

goroutine 6 [syscall, 3 minutes]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:29 +0x41

goroutine 7 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:73 +0x74
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:45 +0x35

goroutine 8 [chan receive, 3 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:100 +0x126
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:46 +0x4d

goroutine 1163151 [chan receive]:
testing.tRunner.func1(0xc00296fa00)
	/usr/local/go/src/testing/testing.go:841 +0x203
testing.tRunner(0xc00296fa00, 0xc00643f280)
	/usr/local/go/src/testing/testing.go:869 +0xca
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x35a

goroutine 16 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc0005712c0)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:152 +0xdd
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:29 +0x57

goroutine 1884307 [select]:
github.com/cockroachdb/cockroach/pkg/storage/closedts/transport.(*Server).Get(0xc001e308a0, 0x149d6787ad80, 0xc0106ba670, 0x149d6787ad80, 0xc0106ba670)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/closedts/transport/server.go:82 +0x2ed
github.com/cockroachdb/cockroach/pkg/storage/closedts/container.delayedServer.Get(0x1, 0x4178f60, 0xc001e308a0, 0x422dfe0, 0xc0106ba670, 0xc02f00010301, 0xc0106ba670)
```

@asubiotto @yuzefovich could you please take a look?

```
panic: flow: unexpected 71680 leftover bytes
```

Hopefully this is reproducible. I'm a bit nervous it snuck into 19.2. See #42705.

SHA: https://github.com/cockroachdb/cockroach/commits/4de1b957fbda6eaa42c8fd7b0fb3278365f7a615

Parameters:
```
TAGS=
GOFLAGS=-parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLogic PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1608053&tab=buildLog

```

panic: flow: unexpected 69632 leftover bytes [recovered]
	panic: panic while executing 1 statements: SELECT _, _ FROM _ UNION ALL SELECT _, _ FROM _ UNION ALL SELECT _, _ FROM _ UNION ALL SELECT _, _ FROM _ UNION ALL SELECT _, _ FROM _ UNION ALL SELECT _, _ FROM _; caused by flow: unexpected 69632 leftover bytes

goroutine 4196204 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc011182880, 0x41e3080, 0xc00153d4c0, 0x33d23c0, 0xc013e605f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:737 +0x330
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc011182880, 0x41e3080, 0xc00153d4c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:448 +0x61
panic(0x33d23c0, 0xc013e605f0)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/util/log.ReportOrPanic(0x41e3140, 0xc013b4bc50, 0xc00bca3500, 0xc00c3004e0, 0x25, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:554 +0x227
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc01d4e9180, 0x41e3140, 0xc013b4bc50, 0x37f8601)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:397 +0x29e
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:384
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*EvalContext).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/eval.go:2795
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cleanup(0xc01d4ef080, 0x41e3140, 0xc013b4bc50)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:437 +0x163
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func7()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:393 +0xcf
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc011182880, 0x41e3140, 0xc018ee9e30, 0xc011182bf0, 0x3, 0x1465966999f8, 0xc010774a80, 0xc00bad8a01, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:841 +0x404
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc011182880, 0x41e3140, 0xc018ee9e30, 0xc011182bf0, 0x1465966999f8, 0xc010774a80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:717 +0x6af
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc011182880, 0x41e3140, 0xc018ee9e30, 0x41fa5c0, 0xc01d194960, 0xc019ccf022, 0x10b, 0x0, 0x6, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:417 +0xc01
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc011182880, 0x41e3140, 0xc018ee9e30, 0x41fa5c0, 0xc01d194960, 0xc019ccf022, 0x10b, 0x0, 0x6, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:98 +0x4f9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc011182880, 0x41e3080, 0xc01d26c200, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1250 +0x1aa2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc011182880, 0x41e3080, 0xc00153d4c0, 0xc0115ad0c0, 0x5400, 0x15000, 0xc0115ad158, 0xc015ac6760, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1179 +0x1c0
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0112d8f00, 0x41e3080, 0xc00153d4c0, 0xc011182880, 0x5400, 0x15000, 0xc0115ad158, 0xc015ac6760, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:450 +0xce
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc00d50a051, 0xc00851cbc0, 0x41e3080, 0xc00153d4c0, 0xc015ac6760, 0xc0112d8f00, 0xc010774400, 0x41f8880, 0xc00851cba0, 0xc016574180, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:584 +0x21c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:513 +0x17b

goroutine 1 [chan receive, 7 minutes]:
testing.(*T).Run(0xc00049f200, 0x391f4b6, 0x9, 0x3a79f68, 0x712016)
```

SHA: https://github.com/cockroachdb/cockroach/commits/60b7b3b2ca511cf9adb30799fcf2b8cffd6223d9

Parameters:
```
TAGS=
GOFLAGS=-parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLogic PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1608924&tab=buildLog

```

=== PAUSE TestLogic/5node-dist-vec/exec_merge_join_dist
=== CONT  TestLogic/5node-dist-vec/exec_merge_join_dist
--- done: testdata/logic_test/exec_merge_join_dist with config 5node-dist-vec: 13 tests, 0 failures
--- progress: testdata/logic_test/exec_hash_join_dist: 16 statements/queries
panic: flow: unexpected 20480 leftover bytes [recovered]
	panic: panic while executing 1 statements: SELECT _._, _._ FROM _ RIGHT HASH JOIN _ ON (_._ = _._) AND (_._ < _) ORDER BY _ LIMIT _; caused by flow: unexpected 20480 leftover bytes

goroutine 4478606 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00b137600, 0x41e1ce0, 0xc01af2bd40, 0x33d1800, 0xc00f219780)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:737 +0x330
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc00b137600, 0x41e1ce0, 0xc01af2bd40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:448 +0x61
panic(0x33d1800, 0xc00f219780)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/util/log.ReportOrPanic(0x41e1da0, 0xc007d85230, 0xc0001c1500, 0xc0021f3a40, 0x25, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:554 +0x227
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc00905d220, 0x41e1da0, 0xc007d85230, 0x41e1d01)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:397 +0x29e
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:384
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*EvalContext).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/eval.go:2795
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cleanup(0xc0052f9340, 0x41e1da0, 0xc007d85230)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:437 +0x163
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func7()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:393 +0xcf
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00b137600, 0x41e1da0, 0xc007d13d70, 0xc00b137970, 0x3, 0x151075e98ce0, 0xc008616f80, 0xc0166e8a01, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:841 +0x404
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00b137600, 0x41e1da0, 0xc007d13d70, 0xc00b137970, 0x151075e98ce0, 0xc008616f80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:717 +0x6af
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00b137600, 0x41e1da0, 0xc007d13d70, 0x41f9220, 0xc001035800, 0xc001a6444d, 0x71, 0x0, 0x2, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:417 +0xc01
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00b137600, 0x41e1da0, 0xc007d13d70, 0x41f9220, 0xc001035800, 0xc001a6444d, 0x71, 0x0, 0x2, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:98 +0x4f9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00b137600, 0x41e1ce0, 0xc0045c3a80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1250 +0x1aa2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00b137600, 0x41e1ce0, 0xc01af2bd40, 0xc011eda340, 0x5400, 0x15000, 0xc011eda3d8, 0xc00c8f8970, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1179 +0x1c0
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0046baf00, 0x41e1ce0, 0xc01af2bd40, 0xc00b137600, 0x5400, 0x15000, 0xc011eda3d8, 0xc00c8f8970, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:450 +0xce
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc016fc2e71, 0xc010ea5500, 0x41e1ce0, 0xc01af2bd40, 0xc00c8f8970, 0xc0046baf00, 0xc008616900, 0x41f74e0, 0xc010ea54c0, 0xc007211380, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:584 +0x21c
```

SHA: https://github.com/cockroachdb/cockroach/commits/f97dc13163020a032b098ef3eb88e4d9f54a04ba

Parameters:
```
TAGS=
GOFLAGS=-parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLogic PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1615116&tab=buildLog

```

panic: flow: unexpected 71680 leftover bytes

goroutine 1952581 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.ReportOrPanic(0x41ea5c0, 0xc0103204b0, 0xc000197500, 0xc012c96450, 0x25, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:554 +0x227
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc00cc4c3c0, 0x41ea5c0, 0xc0103204b0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:397 +0x29e
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:384
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*EvalContext).Stop(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/eval.go:2798
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cleanup(0xc010f566e0, 0x41ea5c0, 0xc0103204b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:437 +0x163
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowScheduler).runFlowNow.func2(0x424c0c0, 0xc01630c180, 0x41ea5c0, 0xc0103204b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow_scheduler.go:104 +0x59
created by github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowScheduler).runFlowNow
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow_scheduler.go:102 +0x280

goroutine 1 [chan receive, 3 minutes]:
testing.(*T).Run(0xc00049e300, 0x39236d6, 0x9, 0x3a7e660, 0x712016)
	/usr/local/go/src/testing/testing.go:917 +0x381
testing.runTests.func1(0xc00049e200)
	/usr/local/go/src/testing/testing.go:1157 +0x78
testing.tRunner(0xc00049e200, 0xc0007c3e10)
	/usr/local/go/src/testing/testing.go:865 +0xc0
testing.runTests(0xc00023c1e0, 0x647ea60, 0x4, 0x4, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x2a9
testing.(*M).Run(0xc0001c6380, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x162
github.com/cockroachdb/cockroach/pkg/sql/logictest.TestMain(0xc0001c6380)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/main_test.go:33 +0xb8
main.main()
	_testmain.go:46 +0x13e

goroutine 6 [syscall, 3 minutes]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:29 +0x41

goroutine 7 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:73 +0x74
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:45 +0x35

goroutine 8 [chan receive, 3 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:100 +0x126
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:46 +0x4d

goroutine 1871081 [sync.Cond.Wait]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:307
sync.runtime_notifyListWait(0xc0042c70d0, 0xc000000037)
	/usr/local/go/src/runtime/sema.go:510 +0xf9
sync.(*Cond).Wait(0xc0042c70c0)
	/usr/local/go/src/sync/cond.go:56 +0x9e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc004902d80, 0x41ea5c0, 0xc00ba32360)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:197 +0x7c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x41ea5c0, 0xc00ba32360)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:166 +0x3e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc010b0fbb0, 0xc00287cd80, 0xc010b0fba0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0xfb
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:190 +0xa8

goroutine 15 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc00033a8c0)
```

--- a/pkg/sql/colexec/routers.go
+++ b/pkg/sql/colexec/routers.go
@@ -40,7 +40,6 @@ type routerOutput interface {
 var defaultRouterOutputBlockedThreshold = int(coldata.BatchSize() * 2)
 
 type routerOutputOp struct {
-	allocator *Allocator
 	// input is a reference to our router.
 	input execinfra.OpNode
 
@@ -56,8 +55,9 @@ type routerOutputOp struct {
 
 	mu struct {
 		syncutil.Mutex
-		cond *sync.Cond
-		done bool
+		allocator *Allocator
+		cond      *sync.Cond
+		done      bool
 		// TODO(asubiotto): Use a ring buffer once we have disk spilling.
 		data      []coldata.Batch
 		numUnread int
@@ -106,13 +106,13 @@ func newRouterOutputOpWithBlockedThresholdAndBatchSize(
 	outputBatchSize int,
 ) *routerOutputOp {
 	o := &routerOutputOp{
-		allocator:           allocator,
 		types:               types,
 		zeroBatch:           allocator.NewMemBatchWithSize(types, 0 /* size */),
 		unblockedEventsChan: unblockedEventsChan,
 		blockedThreshold:    blockedThreshold,
 		outputBatchSize:     outputBatchSize,
 	}
+	o.mu.allocator = allocator
 	o.mu.cond = sync.NewCond(&o.mu)
 	o.mu.data = make([]coldata.Batch, 0, o.blockedThreshold/o.outputBatchSize)
 	return o
@@ -194,11 +194,11 @@ func (o *routerOutputOp) addBatch(batch coldata.Batch, selection []uint16) bool 
 	writeIdx := 0
 	if len(o.mu.data) == 0 {
 		// New output batch.
-		o.mu.data = append(o.mu.data, o.allocator.NewMemBatchWithSize(o.types, o.outputBatchSize))
+		o.mu.data = append(o.mu.data, o.mu.allocator.NewMemBatchWithSize(o.types, o.outputBatchSize))
 	} else {
 		if int(o.mu.data[len(o.mu.data)-1].Length()) == o.outputBatchSize {
 			// No space in last batch, append new output batch.
-			o.mu.data = append(o.mu.data, o.allocator.NewMemBatchWithSize(o.types, o.outputBatchSize))
+			o.mu.data = append(o.mu.data, o.mu.allocator.NewMemBatchWithSize(o.types, o.outputBatchSize))
 		}
 		writeIdx = len(o.mu.data) - 1
 	}
@@ -219,11 +219,11 @@ func (o *routerOutputOp) addBatch(batch coldata.Batch, selection []uint16) bool 
 			numAppended = available
 			// Need to create a new batch to append to in the next o.mu.data slot.
 			// This will be used in the next iteration.
-			o.mu.data = append(o.mu.data, o.allocator.NewMemBatchWithSize(o.types, o.outputBatchSize))
+			o.mu.data = append(o.mu.data, o.mu.allocator.NewMemBatchWithSize(o.types, o.outputBatchSize))
 		}
 
 		for i, t := range o.types {
-			o.allocator.Append(
+			o.mu.allocator.Append(
 				dst.ColVec(i),
 				coldata.SliceArgs{
 					ColType:   t,

--- a/pkg/sql/colflow/vectorized_flow.go
+++ b/pkg/sql/colflow/vectorized_flow.go
@@ -40,10 +40,10 @@ type vectorizedFlow struct {
 	// operatorConcurrency is set if any operators are executed in parallel.
 	operatorConcurrency bool
 
-	// streamingMemAccount is the memory account that is tracking the static
+	// streamingMemAccounts are the memory accounts that are tracking the static
 	// memory usage of the whole vectorized flow as well as all dynamic memory of
 	// the streaming components.
-	streamingMemAccount *mon.BoundAccount
+	streamingMemAccounts []*mon.BoundAccount
 
 	// bufferingMemMonitors are the memory monitors of the buffering components.
 	bufferingMemMonitors []*mon.BytesMonitor
@@ -73,8 +73,6 @@ func (f *vectorizedFlow) Setup(
 ) error {
 	f.SetSpec(spec)
 	log.VEventf(ctx, 1, "setting up vectorize flow %s", f.ID.Short())
-	streamingMemAccount := f.EvalCtx.Mon.MakeBoundAccount()
-	f.streamingMemAccount = &streamingMemAccount
 	recordingStats := false
 	if sp := opentracing.SpanFromContext(ctx); sp != nil && tracing.IsRecording(sp) {
 		recordingStats = true
@@ -89,18 +87,21 @@ func (f *vectorizedFlow) Setup(
 		f.GetFlowCtx().Cfg.NodeDialer,
 		f.GetID(),
 	)
-	_, err := creator.setupFlow(ctx, f.GetFlowCtx(), spec.Processors, &streamingMemAccount, opt)
+	_, err := creator.setupFlow(ctx, f.GetFlowCtx(), spec.Processors, opt)
 	if err == nil {
 		f.operatorConcurrency = creator.operatorConcurrency
+		f.streamingMemAccounts = append(f.streamingMemAccounts, creator.streamingMemAccounts...)
 		f.bufferingMemMonitors = append(f.bufferingMemMonitors, creator.bufferingMemMonitors...)
 		f.bufferingMemAccounts = append(f.bufferingMemAccounts, creator.bufferingMemAccounts...)
 		log.VEventf(ctx, 1, "vectorized flow setup succeeded")
 		return nil
 	}
-	f.streamingMemAccount.Close(ctx)
 	// It is (theoretically) possible that some of the memory monitoring
 	// infrastructure was created even in case of an error, and we need to clean
 	// that up.
+	for _, memAcc := range creator.streamingMemAccounts {
+		memAcc.Close(ctx)
+	}
 	for _, memAcc := range creator.bufferingMemAccounts {
 		memAcc.Close(ctx)
 	}
@@ -125,7 +126,9 @@ func (f *vectorizedFlow) Release() {
 // Cleanup is part of the Flow interface.
 func (f *vectorizedFlow) Cleanup(ctx context.Context) {
 	// This cleans up all the memory monitoring of the vectorized flow.
-	f.streamingMemAccount.Close(ctx)
+	for _, memAcc := range f.streamingMemAccounts {
+		memAcc.Close(ctx)
+	}
 	for _, memAcc := range f.bufferingMemAccounts {
 		memAcc.Close(ctx)
 	}
@@ -276,6 +279,9 @@ type vectorizedFlowCreator struct {
 	leaves []execinfra.OpNode
 	// operatorConcurrency is set if any operators are executed in parallel.
 	operatorConcurrency bool
+	// streamingMemAccounts contains all memory accounts of the non-buffering
+	// components in the vectorized flow.
+	streamingMemAccounts []*mon.BoundAccount
 	// bufferingMemMonitors contains all memory monitors of the buffering
 	// components in the vectorized flow.
 	bufferingMemMonitors []*mon.BytesMonitor
@@ -307,19 +313,29 @@ func newVectorizedFlowCreator(
 	}
 }
 
+// newStreamingMemAccount creates a new memory account bound to the monitor in
+// flowCtx and accumulates it into streamingMemAccounts slice.
+func (s *vectorizedFlowCreator) newStreamingMemAccount(
+	flowCtx *execinfra.FlowCtx,
+) *mon.BoundAccount {
+	streamingMemAccount := flowCtx.EvalCtx.Mon.MakeBoundAccount()
+	s.streamingMemAccounts = append(s.streamingMemAccounts, &streamingMemAccount)
+	return &streamingMemAccount
+}
+
 // setupRemoteOutputStream sets up an Outbox that will operate according to
 // the given StreamEndpointSpec. It will also drain all MetadataSources in the
 // metadataSourcesQueue.
 func (s *vectorizedFlowCreator) setupRemoteOutputStream(
 	ctx context.Context,
+	flowCtx *execinfra.FlowCtx,
 	op colexec.Operator,
 	outputTyps []coltypes.T,
 	stream *execinfrapb.StreamEndpointSpec,
 	metadataSourcesQueue []execinfrapb.MetadataSource,
-	streamingMemAccount *mon.BoundAccount,
 ) (execinfra.OpNode, error) {
 	outbox, err := s.remoteComponentCreator.newOutbox(
-		colexec.NewAllocator(ctx, streamingMemAccount),
+		colexec.NewAllocator(ctx, s.newStreamingMemAccount(flowCtx)),
 		op, outputTyps, metadataSourcesQueue,
 	)
 	if err != nil {
@@ -360,7 +376,6 @@ func (s *vectorizedFlowCreator) setupRouter(
 	outputTyps []coltypes.T,
 	output *execinfrapb.OutputRouterSpec,
 	metadataSourcesQueue []execinfrapb.MetadataSource,
-	streamingMemAccount *mon.BoundAccount,
 ) error {
 	if output.Type != execinfrapb.OutputRouterSpec_BY_HASH {
 		return errors.Errorf("vectorized output router type %s unsupported", output.Type)
@@ -397,7 +412,7 @@ func (s *vectorizedFlowCreator) setupRouter(
 			return errors.Errorf("unexpected sync response output when setting up router")
 		case execinfrapb.StreamEndpointSpec_REMOTE:
 			if _, err := s.setupRemoteOutputStream(
-				ctx, op, outputTyps, stream, metadataSourcesQueue, streamingMemAccount,
+				ctx, flowCtx, op, outputTyps, stream, metadataSourcesQueue,
 			); err != nil {
 				return err
 			}
@@ -438,9 +453,9 @@ func (s *vectorizedFlowCreator) setupRouter(
 // calling DrainMeta.
 func (s *vectorizedFlowCreator) setupInput(
 	ctx context.Context,
+	flowCtx *execinfra.FlowCtx,
 	input execinfrapb.InputSyncSpec,
 	opt flowinfra.FuseOpt,
-	streamingMemAccount *mon.BoundAccount,
 ) (op colexec.Operator, _ []execinfrapb.MetadataSource, _ error) {
 	inputStreamOps := make([]colexec.Operator, 0, len(input.Streams))
 	metaSources := make([]execinfrapb.MetadataSource, 0, len(input.Streams))
@@ -460,7 +475,10 @@ func (s *vectorizedFlowCreator) setupInput(
 			if err != nil {
 				return nil, nil, err
 			}
-			inbox, err := s.remoteComponentCreator.newInbox(colexec.NewAllocator(ctx, streamingMemAccount), typs, inputStream.StreamID)
+			inbox, err := s.remoteComponentCreator.newInbox(
+				colexec.NewAllocator(ctx, s.newStreamingMemAccount(flowCtx)),
+				typs, inputStream.StreamID,
+			)
 			if err != nil {
 				return nil, nil, err
 			}
@@ -496,7 +514,8 @@ func (s *vectorizedFlowCreator) setupInput(
 		}
 		if input.Type == execinfrapb.InputSyncSpec_ORDERED {
 			op = colexec.NewOrderedSynchronizer(
-				colexec.NewAllocator(ctx, streamingMemAccount), inputStreamOps, typs, execinfrapb.ConvertToColumnOrdering(input.Ordering),
+				colexec.NewAllocator(ctx, s.newStreamingMemAccount(flowCtx)),
+				inputStreamOps, typs, execinfrapb.ConvertToColumnOrdering(input.Ordering),
 			)
 		} else {
 			if opt == flowinfra.FuseAggressively {
@@ -536,7 +555,6 @@ func (s *vectorizedFlowCreator) setupOutput(
 	op colexec.Operator,
 	opOutputTypes []coltypes.T,
 	metadataSourcesQueue []execinfrapb.MetadataSource,
-	streamingMemAccount *mon.BoundAccount,
 ) error {
 	output := &pspec.Output[0]
 	if output.Type != execinfrapb.OutputRouterSpec_PASS_THROUGH {
@@ -549,7 +567,6 @@ func (s *vectorizedFlowCreator) setupOutput(
 			// Pass in a copy of the queue to reset metadataSourcesQueue for
 			// further appends without overwriting.
 			metadataSourcesQueue,
-			streamingMemAccount,
 		)
 	}
 
@@ -583,7 +600,7 @@ func (s *vectorizedFlowCreator) setupOutput(
 				},
 			)
 		}
-		outbox, err := s.setupRemoteOutputStream(ctx, op, opOutputTypes, outputStream, metadataSourcesQueue, streamingMemAccount)
+		outbox, err := s.setupRemoteOutputStream(ctx, flowCtx, op, opOutputTypes, outputStream, metadataSourcesQueue)
 		if err != nil {
 			return err
 		}
@@ -636,7 +653,6 @@ func (s *vectorizedFlowCreator) setupFlow(
 	ctx context.Context,
 	flowCtx *execinfra.FlowCtx,
 	processorSpecs []execinfrapb.ProcessorSpec,
-	streamingMemAccount *mon.BoundAccount,
 	opt flowinfra.FuseOpt,
 ) (leaves []execinfra.OpNode, err error) {
 	streamIDToSpecIdx := make(map[execinfrapb.StreamID]int)
@@ -678,7 +694,7 @@ func (s *vectorizedFlowCreator) setupFlow(
 		metadataSourcesQueue := make([]execinfrapb.MetadataSource, 0, 1)
 		inputs = inputs[:0]
 		for i := range pspec.Input {
-			input, metadataSources, err := s.setupInput(ctx, pspec.Input[i], opt, streamingMemAccount)
+			input, metadataSources, err := s.setupInput(ctx, flowCtx, pspec.Input[i], opt)
 			if err != nil {
 				return nil, err
 			}
@@ -687,7 +703,7 @@ func (s *vectorizedFlowCreator) setupFlow(
 		}
 
 		result, err := colexec.NewColOperator(
-			ctx, flowCtx, pspec, inputs, streamingMemAccount,
+			ctx, flowCtx, pspec, inputs, s.newStreamingMemAccount(flowCtx),
 			false, /* useStreamingMemAccountForBuffering */
 		)
 		// Even when err is non-nil, it is possible that the buffering memory
@@ -707,7 +723,10 @@ func (s *vectorizedFlowCreator) setupFlow(
 			!result.IsStreaming {
 			return nil, errors.Errorf("non-streaming operator encountered when vectorize=auto")
 		}
-		if err = streamingMemAccount.Grow(ctx, int64(result.InternalMemUsage)); err != nil {
+		// We created a streaming memory account when calling NewColOperator above,
+		// so there is definitely at least one memory account, and it doesn't
+		// matter which one we grow.
+		if err = s.streamingMemAccounts[0].Grow(ctx, int64(result.InternalMemUsage)); err != nil {
 			return nil, errors.Wrapf(err, "not enough memory to setup vectorized plan")
 		}
 		metadataSourcesQueue = append(metadataSourcesQueue, result.MetadataSources...)
@@ -735,7 +754,7 @@ func (s *vectorizedFlowCreator) setupFlow(
 			return nil, err
 		}
 		if err = s.setupOutput(
-			ctx, flowCtx, pspec, op, opOutputTypes, metadataSourcesQueue, streamingMemAccount,
+			ctx, flowCtx, pspec, op, opOutputTypes, metadataSourcesQueue,
 		); err != nil {
 			return nil, err
 		}
@@ -919,9 +938,10 @@ func SupportsVectorized(
 	)
 	memoryMonitor.Start(ctx, nil, mon.MakeStandaloneBudget(math.MaxInt64))
 	defer memoryMonitor.Stop(ctx)
-	acc := memoryMonitor.MakeBoundAccount()
-	defer acc.Close(ctx)
 	defer func() {
+		for _, memAcc := range creator.streamingMemAccounts {
+			memAcc.Close(ctx)
+		}
 		for _, memAcc := range creator.bufferingMemAccounts {
 			memAcc.Close(ctx)
 		}
@@ -930,7 +950,7 @@ func SupportsVectorized(
 		}
 	}()
 	if vecErr := execerror.CatchVectorizedRuntimeError(func() {
-		leaves, err = creator.setupFlow(ctx, flowCtx, processorSpecs, &acc, fuseOpt)
+		leaves, err = creator.setupFlow(ctx, flowCtx, processorSpecs, fuseOpt)
 	}); vecErr != nil {
 		return leaves, vecErr
 	}

--- a/pkg/sql/colflow/vectorized_flow_test.go
+++ b/pkg/sql/colflow/vectorized_flow_test.go
@@ -218,9 +218,12 @@ func TestDrainOnlyInputDAG(t *testing.T) {
 		execinfrapb.FlowID{},
 	)
 
-	acc := evalCtx.Mon.MakeBoundAccount()
-	defer acc.Close(ctx)
-	_, err := vfc.setupFlow(ctx, &f.FlowCtx, procs, &acc, flowinfra.FuseNormally)
+	_, err := vfc.setupFlow(ctx, &f.FlowCtx, procs, flowinfra.FuseNormally)
+	defer func() {
+		for _, memAcc := range vfc.streamingMemAccounts {
+			memAcc.Close(ctx)
+		}
+	}()
 	require.NoError(t, err)
 
 	// Verify that an outbox was actually created.

--- a/pkg/util/mon/bytes_usage.go
+++ b/pkg/util/mon/bytes_usage.go
@@ -632,10 +632,10 @@ func (mm *BytesMonitor) releaseBytes(ctx context.Context, sz int64) {
 	mm.mu.Lock()
 	defer mm.mu.Unlock()
 	if mm.mu.curAllocated < sz {
-		sz = mm.mu.curAllocated
 		log.ReportOrPanic(ctx, &mm.settings.SV,
 			"%s: no bytes to release, current %d, free %d",
 			mm.name, mm.mu.curAllocated, sz)
+		sz = mm.mu.curAllocated
 	}
 	mm.mu.curAllocated -= sz
 	if mm.curBytesCount != nil {
@@ -647,7 +647,7 @@ func (mm *BytesMonitor) releaseBytes(ctx context.Context, sz int64) {
 		// We avoid VEventf here because we want to avoid computing the
 		// trace string if there is nothing to log.
 		log.Infof(ctx, "%s: now at %d bytes (-%d) - %s",
-			mm.name, mm.mu.curAllocated, sz, util.GetSmallTrace(3))
+			mm.name, mm.mu.curAllocated, sz, util.GetSmallTrace(5))
 	}
 }
 

